server/mscalendar/settings_daily_summary.go

```diff
@@ -0,0 +1,259 @@
+	optionsM    []string
+	optionsAPM  []string
+	store       settingspanel.SettingStore
+	getTimezone func(userID string) (string, error)
+}
+
+func NewDailySummarySetting(inStore settingspanel.SettingStore, getTimezone func(userID string) (string, error)) settingspanel.Setting {
+	os := &dailySummarySetting{
+		title:       "Daily Summary",
+		description: "When do you want to receive the daily summary?\n If you update this setting, it will automatically update to your the timezone currently set on your calendar.",
+		id:          store.DailySummarySettingID,
+		dependsOn:   "",
+		store:       inStore,
+		getTimezone: getTimezone,
+	}
+	os.optionsH = []string{}
+	for i := 0; i < 12; i++ {
+		os.optionsH = append(os.optionsH, fmt.Sprintf("%d", i))
+	}
+
+	os.optionsM = []string{}
+	for i := 0; i < 4; i++ {
+		os.optionsM = append(os.optionsM, fmt.Sprintf("%02d", i*15))
+	}
+
+	os.optionsAPM = []string{"AM", "PM"}
+
+	return os
+}
+
+func (s *dailySummarySetting) Set(userID string, value interface{}) error {
+	_, ok := value.(string)
+	if !ok {
+		return errors.New("trying to set Daily Summary Setting without a string value")
+	}
+	err := s.store.SetSetting(userID, s.id, value)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *dailySummarySetting) Get(userID string) (interface{}, error) {
+	value, err := s.store.GetSetting(userID, s.id)
+	if err != nil {
+		return nil, err
+	}
+
+	_, ok := value.(*store.DailySummaryUserSettings)
+	if !ok {
+		return nil, errors.New("current value is not a Daily Summary Setting")
+	}
+
+	return value, nil
+}
+
+func (s *dailySummarySetting) GetID() string {
+	return s.id
+}
+
+func (s *dailySummarySetting) GetTitle() string {
+	return s.title
+}
+
+func (s *dailySummarySetting) GetDescription() string {
+	return s.description
+}
+
+func (s *dailySummarySetting) GetDependency() string {
+	return s.dependsOn
+}
+
+func (s *dailySummarySetting) GetSlackAttachments(userID, settingHandler string, disabled bool) (*model.SlackAttachment, error) {
+	title := fmt.Sprintf("Setting: %s", s.title)
+	currentValueMessage := "Disabled"
+
+	actions := []*model.PostAction{}
+	if !disabled {
+		dsumRaw, err := s.Get(userID)
+		if err != nil {
+			return nil, err
+		}
+		dsum := dsumRaw.(*store.DailySummaryUserSettings)
+
+		currentH := "8"
+		currentM := "00"
+		currentAPM := "AM"
+		fullTime := "8:00AM"
+		currentEnable := false
+		currentTextValue := "Not set."
+
+		if dsum != nil {
+			fullTime = dsum.PostTime
+			currentEnable = dsum.Enable
+			splitted := strings.Split(fullTime, ":")
+			currentH = splitted[0]
+			currentM = splitted[1][:2]
+			currentAPM = splitted[1][2:]
+			enableText := "Disabled"
+			if currentEnable {
+				enableText = "Enabled"
+			}
+			currentTextValue = fmt.Sprintf("%s (%s) (%s)", dsum.PostTime, dsum.Timezone, enableText)
+		}
+
+		timezone, err := s.getTimezone(userID)
+		if err != nil {
+			if dsum != nil {
+				timezone = dsum.Timezone
+			} else {
+				timezone = "UTC"
```

Comment:
I'm not sure what benefit the user gets in setting their settings with the wrong timezone. 1/5 Instead of falling back on UTC, return an error here saying that we couldn't get their timezone from Microsoft.